src/main/java/controller/EngineController.java

```diff
@@ -0,0 +1,182 @@
+package controller;
+
+import exception.CarRentalException;
+import lombok.RequiredArgsConstructor;
+import models.cars.Engine;
+import models.cars.FuelType;
+import models.cars.TransmissionType;
+import models.session.Session;
+import services.engine.EngineCharacteristics;
+import services.engine.EngineService;
+import services.engine.FuelTypeService;
+import services.engine.TransmissionTypeService;
+import services.session.SessionService;
+import services.user.UserService;
+
+import java.util.List;
+import java.util.Scanner;
+
+import static exception.ExceptionHandler.handleException;
+
+@RequiredArgsConstructor
+public class EngineController {
+    private final EngineService engineService;
+    private final SessionService sessionService;
+    private final UserService userService;
+    private final FuelTypeService fuelTypeService;
+    private final TransmissionTypeService transmissionTypeService;
+    private final Scanner scanner;
+
+
+    public void programInterface() {
+        handleException(() -> {
+            Session session = sessionService.getActive();
+            while (true) {
+                if (userService.isManager(session.getUserId()) && sessionService.isUserAuthenticated()) {
```

Comment:
I think you need to check if user authenticated before getting the session. Because if user is not authenticated you will get an exception